internal/app/machined/pkg/controllers/network/operator/wglan.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	ip, err := wgEUI64(prefix)
+	if err != nil {
+		logger.Warn("failed to generate local IP address", zap.Error(err))
+
+		return nil
+	}
+
+	pubKey := privKey.PublicKey().String()
+
+	cfg := &wglan.Config{
+		ClusterID:    clusterID,
+		DiscoveryURL: discoveryURL,
+		IP:           ip,
+		LinkName:     linkName,
+		Nodename:     nodenameFunc,
+		PublicKey:    pubKey,
+		RoutingTable: constants.WireguardDefaultRoutingTable,
+		Subnet:       prefix,
+	}
+
+	return &WgLAN{
+		Config:     cfg,
+		logger:     logger,
+		fwMark:     constants.WireguardDefaultFirewallMark,
+		privateKey: privateKey,
+		listenPort: constants.WireguardDefaultPort,
+		db:         new(wglan.PeerDB),
+	}
+}
+
+// Prefix returns unique operator prefix which gets prepended to each spec.
+func (o *WgLAN) Prefix() string {
+	return fmt.Sprintf("wglan/%s", o.Config.LinkName)
+}
+
+// Run the operator loop.
+func (o *WgLAN) Run(ctx context.Context, notifyCh chan<- struct{}) {
+	var rulesManager *wglan.RulesManager
+
+	var peerManager *wglan.PeerManager
+
+	for ctx.Err() == nil {
+		time.Sleep(time.Second)
+
+		rulesManager = new(wglan.RulesManager)
+
+		if err := rulesManager.Run(ctx, o.logger, o.db); err != nil {
+			o.logger.Warn("failed to start rules manager", zap.Error(err))
+
+			continue
+		}
+
+		peerManager = wglan.NewPeerManager(o.Config, o.db)
+
+		if err := peerManager.Run(ctx, o.logger, notifyCh); err != nil {
+			o.logger.Warn("failed to start peer manager", zap.Error(err))
+		}
+	}
+}
+
+// AddressSpecs implements Operator interface.
+func (o *WgLAN) AddressSpecs() []network.AddressSpecSpec {
+	if o == nil || o.Config == nil {
+		return nil
+	}
+
+	return []network.AddressSpecSpec{
+		{
+			Address:         o.Config.IP,
+			LinkName:        o.Config.LinkName,
+			Family:          unix.AF_INET6,
+			Scope:           unix.RT_SCOPE_UNIVERSE,
+			Flags:           0,
+			AnnounceWithARP: false,
+			ConfigLayer:     network.ConfigOperator,
+		},
+	}
+}
+
+// LinkSpecs implements Operator interface.
+func (o *WgLAN) LinkSpecs() []network.LinkSpecSpec {
+	if o == nil || o.Config == nil {
+		return nil
+	}
+
+	peerList := o.getPeers(o.listenPort)
+
+	return []network.LinkSpecSpec{
+		{
+			Name:       o.Config.LinkName,
+			Logical:    true,
+			Up:         true,
+			Kind:       network.LinkKindWireguard,
+			Type:       nethelpers.LinkNetrom,
+			VLAN:       network.VLANSpec{},
+			BondMaster: network.BondMasterSpec{},
+			Wireguard: network.WireguardSpec{
+				PrivateKey:   o.privateKey,
+				ListenPort:   int(o.listenPort),
+				FirewallMark: o.fwMark,
+				Peers:        peerList,
+			},
+			ConfigLayer: network.ConfigOperator,
+		},
+	}
+}
+
+// RouteSpecs implements Operator interface.
+func (o *WgLAN) RouteSpecs() []network.RouteSpecSpec {
+	if o == nil || o.Config == nil {
+		return nil
+	}
+
+	return []network.RouteSpecSpec{
+		{
+			Family:      unix.AF_INET,
+			Destination: netaddr.MustParseIPPrefix("0.0.0.0/0"),
```

Comment:
zero values should be left out (just use `netaddr.IPPrefix{}` or skip the field), this is our convention so far

Reply:
this leads to route flapping back and forth right now